src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStringConcat.cs

```diff
@@ -205,79 +205,115 @@ static void Main()
             comp.VerifyDiagnostics();
             comp.VerifyIL("Test.Main", @"
 {
-  // Code size      187 (0xbb)
-  .maxstack  4
+  // Code size      259 (0x103)
+  .maxstack  5
   IL_0000:  ldsfld     ""string Test.S""
   IL_0005:  ldstr      ""AB""
   IL_000a:  ldsfld     ""string Test.S""
   IL_000f:  call       ""string string.Concat(string, string, string)""
   IL_0014:  call       ""void System.Console.WriteLine(string)""
   IL_0019:  ldsfld     ""object Test.O""
-  IL_001e:  ldstr      ""AB""
-  IL_0023:  ldsfld     ""object Test.O""
-  IL_0028:  call       ""string string.Concat(object, object, object)""
-  IL_002d:  call       ""void System.Console.WriteLine(string)""
-  IL_0032:  ldc.i4.6
-  IL_0033:  newarr     ""object""
-  IL_0038:  dup
-  IL_0039:  ldc.i4.0
-  IL_003a:  ldsfld     ""string Test.S""
-  IL_003f:  stelem.ref
-  IL_0040:  dup
-  IL_0041:  ldc.i4.1
-  IL_0042:  ldstr      ""AB""
-  IL_0047:  stelem.ref
-  IL_0048:  dup
-  IL_0049:  ldc.i4.2
-  IL_004a:  ldsfld     ""string Test.S""
-  IL_004f:  stelem.ref
+  IL_001e:  dup
+  IL_001f:  brtrue.s   IL_0025
+  IL_0021:  pop
+  IL_0022:  ldnull
```

Comment:
wondering if someone who knows IL better than me can comment: does it make sense to `brtrue`, and in the 'false' case do a `pop` then `ldnull`? Don't you already have a null on the stack in this case?

Reply:
Seems like it's just how we emit conditional access, so I don't necessarily think this should change in this PR.

---
In reply to: [278354559](https://github.com/dotnet/roslyn/pull/35006#discussion_r278354559) [](ancestors = 278354559)

Reply:
I actually tried making this change to the conditional access generation (I had the same thought). The problem seems to be that we've got the wrong sort of null in the stack.

Reply:
Here's a thought: if we call a `string.Concat` overload which takes `objects`, then perhaps we can leave the `null` on the stack, rather than `pop ldnull`. The downside is that `string.Concat` will go and call `ToString` on a bunch of strings.